### PR TITLE
[SERVER-1469] server 3.3 - adding docs for nomad-autoscaler on server

### PR DESCRIPTION
**Pull request by soulchips**
# Description
We will be releasing support for nomad's autoscaler service with server `3.3.0`. This PR contains a rough draft of the documentation that we hope the docs team can make pretty :)

# Reasons
https://circleci.atlassian.net/browse/SERVER-1469

### CODE REVIEW

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (ryanwohara) [inline jekyll/_cci2/server-3-install-build-services.adoc:171]:
```suggestion
Nomad provides a utility to automatically scale up or down your nomad clients provided your clients are managed by a cloud provider's autoscaling resource. With Nomad Autoscaler, you only need to provide permission for the utility to manage your autoscaling resource and where it is located. You can enable this resource via KOTS, which will deploy the Nomad Autoscaler service along with your Nomad servers. Below we will go through how to set up Nomad Autoscaler for your provider.
```

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (ryanwohara) [inline jekyll/_cci2/server-3-install-build-services.adoc:173]:
```suggestion
If you do not require this service then feel free to jump to the *Save config* button to update your installation and re-deploy server.
```

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (ryanwohara) [inline jekyll/_cci2/server-3-install-build-services.adoc:177]:
```suggestion
  * Our link:https://github.com/CircleCI-Public/server-terraform/tree/main/nomad-aws[nomad module] creates an IAM user and outputs the keys if you set variable `nomad_auto_scaler = true`. You may reference the example in the link for more details. If you've already created the clients, you can simply update the variable and run `terraform apply`. The created user's access key and secret will be available in Terraform's output.
```

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (ryanwohara) [inline jekyll/_cci2/server-3-install-build-services.adoc:205]:
```suggestion
. Set Max Node Count* - This will overwrite what is currently set as the max for you ASG. It is recommended to keep this value and what was set in your Terraform as the same.
```

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (ryanwohara) [inline jekyll/_cci2/server-3-install-build-services.adoc:209]:
```suggestion
. Add the Nomad Autoscaler user's access key
```

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (ryanwohara) [inline jekyll/_cci2/server-3-install-build-services.adoc:227]:
```suggestion
* The maximum and minimum nomad client count will overwrite the corresponding values set when you created your autoscaling group or managed instance group. It is recommended that you keep these values and those used in your Terraform the same so that the two don't compete.
```

**Comment by soulchips** [comment]
thanks for the quick review @ryanwohara !!!

**Review by ryanwohara — APPROVED** [review]

**Review by devinmarieb — APPROVED** [review]
Thanks, I'll leave it to @rosieyohannan for final approval from the docs team 🙌

File: jekyll/_cci2/server-3-install-build-services.adoc
Comment (rosieyohannan) [inline jekyll/_cci2/server-3-install-build-services.adoc:177]: It feels like we should talk about this step earlier so if people know they want to autoscale they can do this at the point of  deploying their nomad clients for the first time.

Also, this feels like something customers want to do after the initial install so I will make a server 3 specific guide to nomad to include this info (up to now we've just been using the nomad guide from v2 - https://circleci.com/docs/2.0/nomad/) - I'll do this in a separate PR once we've merged this into the 3.3 branch though. Let me know if that doesn't sound right.

Comment (soulchips) [reply jekyll/_cci2/server-3-install-build-services.adoc:161]: I wasnt sure myself. My thinking though is for first install, they may want to focus on getting things running and can return to autoscaling. its an optional step and does no harm to do this later. 
However I dont have a strong opinion here and am guided by the your experience.

Comment (rosieyohannan) [reply jekyll/_cci2/server-3-install-build-services.adoc:161]: I put a note in above to flag this section up. Let's leave as is for now, can always rearrange later.

**Pull request merged**
